ui/apps/platform/src/Components/CompoundSearchFilter/components/SearchFilterAutocomplete.tsx

```diff
@@ -0,0 +1,288 @@
+
+                break;
+            case 'Tab':
+            case 'Escape':
+                setIsOpen(false);
+                setActiveItem(null);
+                break;
+            case 'ArrowUp':
+            case 'ArrowDown':
+                event.preventDefault();
+                handleMenuArrowKeys(event.key);
+                break;
+            default:
+        }
+    };
+
+    const toggle = (toggleRef: React.Ref<MenuToggleElement>) => (
+        <MenuToggle
+            ref={toggleRef}
+            variant="typeahead"
+            onClick={onToggleClick}
+            isExpanded={isOpen}
+            isFullWidth
+            aria-labelledby={`${textLabel} menu toggle`}
+        >
+            <TextInputGroup isPlain>
+                <TextInputGroupMain
+                    value={value}
+                    onClick={onToggleClick}
+                    onChange={onTextInputChange}
+                    onKeyDown={onInputKeyDown}
+                    id="typeahead-select-input"
+                    autoComplete="off"
+                    innerRef={textInputRef}
+                    placeholder={textLabel}
+                    {...(activeItem && { 'aria-activedescendant': activeItem })}
+                    role="combobox"
+                    isExpanded={isOpen}
+                    aria-controls="select-typeahead-listbox"
+                    aria-label={textLabel}
+                />
+
+                <TextInputGroupUtilities>
+                    {!!value && (
+                        <Button
+                            variant="plain"
+                            onClick={() => {
+                                onChange('');
+                                setFilterValue('');
+                                textInputRef?.current?.focus();
+                            }}
+                            aria-label="Clear input value"
+                        >
+                            <TimesIcon aria-hidden />
+                        </Button>
+                    )}
+                </TextInputGroupUtilities>
+            </TextInputGroup>
+        </MenuToggle>
+    );
+
+    return (
+        <>
+            <Select
+                aria-label={`${textLabel} select menu`}
+                isOpen={isOpen}
+                selected={value}
+                onSelect={onSelect}
+                onOpenChange={() => {
+                    setIsOpen(false);
+                }}
+                toggle={toggle}
+            >
+                <SelectList id="select-typeahead-listbox">
+                    {selectOptions.map((option, index) => (
+                        <SelectOption
+                            key={option.value || option.children}
+                            isFocused={focusedItemIndex === index}
+                            className={option.className}
+                            onClick={() => onChange(option.value)}
+                            id={`select-typeahead-${option.value.replace(' ', '-')}`}
```

Comment:
Probably a nit, but it seems within the realm of possibilities that a set of autocomplete options could be returned that includes both an item with a space delimiter and an item with a hyphen delimiter. e.g.
```
apk tools
apk-tools
```
which would result in duplicate ids here. Honestly though, looking at the spreadsheet I can't seem to find a place where we would expect any value to contain a space. It seems like these are normalized on the backend for the most part. Either way, what do you think of the following:

1. Instead of `-`, use a much less likely string literal as the replacement ala `value.replace(' ' , '-space-')`
2. Move the construction of the id into a helper function so that the replacement does not get out of sync in the two places this is constructed

Reply:
That's a good point. I wonder if the `id` is even worth it. In our tests, we usually search by text, so we might consider removing this (unless there is a good reason not to)

Reply:
I think your suggestion is fine here. I don't imagine us using the `id` for testing purposes. We usually search by the text. In this case, we're just using the `id` to highlight what is active. It doesn't need to too human-readable.